src/makeCalculator.js

```diff
@@ -4,7 +4,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  const calculator = {
+    add(num) {
+      result += num;
+    },
+    subtract(num) {
+      result -= num;
+    },
+    multiply(num) {
+      result *= num;
+    },
+    divide(num) {
+      result /= num;
+    },
+    reset() {
+      result = 0;
+
+      return this;
+    },
+    operate(callback, num) {
+      callback(num);
+
+      return this;
+    },
+    get result() {
+      return result;
+    },
```

Comment:
The `result` property is implemented as a getter, which is not necessary and could lead to confusion since `result` is supposed to be a direct property of the calculator object. It should be a regular property that holds the current result value.